ci-operator/step-registry/baremetalds/assisted/operator/setup/baremetalds-assisted-operator-setup-commands.sh

```diff
@@ -0,0 +1,179 @@
+  name: assisted-installer
+  labels:
+    name: assisted-installer
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  labels:
+    app: postgres
+  name: postgres-pv-claim
+  namespace: assisted-installer
+spec:
+  storageClassName: "fs-lso"
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 20Gi
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  labels:
+    app: scality
+  name: scality-pv-claim
+  namespace: assisted-installer
+spec:
+  storageClassName: "fs-lso"
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 40Gi
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: assisted-service-catalog
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: quay.io/ocpmetal/assisted-service-index:latest
+EOCR
+
+SECRET=\$(oc get secret --field-selector type=kubernetes.io/dockercfg -o name | grep default)
+PULL_SECRET=\$(oc get \$SECRET -o json | jq -r '.data[".dockercfg"]' | base64 -d)
+
+echo "Creating pull-secret..."
+cat <<EOCR | oc create -f -
+apiVersion: v1
+kind: Secret
+type: kubernetes.io/dockerconfigjson
+metadata:
+  name: pull-secret
+  namespace: assisted-installer
+stringData:
+  .dockerconfigjson: '\$PULL_SECRET'
+EOCR
+
+echo "Installing assisted-installer operator..."
+cat <<EOCR | oc create -f -
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+    name: assisted-installer-group
+    namespace: assisted-installer
+spec:
+  targetNamespaces:
+    - assisted-installer
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: assisted-service-operator
+  namespace: assisted-installer
+spec:
+  channel: alpha
+  installPlanApproval: Automatic
+  name: assisted-service-operator
+  source: assisted-service-catalog
+  sourceNamespace: openshift-marketplace
+  startingCSV: assisted-service-operator.v0.0.1
+  config:
+    env:
+      - name: DEPLOY_TARGET
+        value: "onprem"
+EOCR
+
+wait_for_crd "agentserviceconfigs.agent-install.openshift.io"
+
+cat <<EOCR | oc create -f -
+apiVersion: agent-install.openshift.io/v1beta1
+kind: AgentServiceConfig
+metadata:
+ namespace: assisted-installer
+ name: agent
+spec:
+ databaseStorage:
+  storageClassName: assisted-service
+  accessModes:
+  - ReadWriteOnce
+  resources:
+   requests:
+    storage: 8Gi
+ filesystemStorage:
+  storageClassName: assisted-service
+  accessModes:
+  - ReadWriteOnce
+  resources:
+   requests:
+    storage: 8Gi
+EOCR
+
+echo "Waiting the assisted-service to get installed..."
+for i in {1..40}; do
+  pods=$(oc get pod -l app=assisted-service -n assisted-installer 2>&1 > /dev/null)
+  if [[ \${pods} != "No resources found"* ]]; then
+    break
+  fi
+
+  sleep 10
+done
+
+echo "all resources:"
+oc get all -n assisted-installer
+
+echo "waiting for assisted-service pod to be ready..."
+oc wait -n assisted-installer --for=condition=Ready pod -l app=assisted-service --timeout=90s
```

Comment:
What's the difference between the two waiting?

Reply:
hmm, waiting for creation of the object and waiting for the pod to become ready
but maybe I'll replace with waiting for the CSV using the code you sent to me